clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -77,6 +84,13 @@
     private static final String SUBSCRIPTION_EXCEPTION_MESSAGE =
         "Subscription to topics, partitions and pattern are mutually exclusive";
 
+    private void confirmSubscriptionType(SubscriptionType type) {
+        if (this.subscriptionType.equals(SubscriptionType.NONE))
```

Comment:
Another nit: you can use == to test enums. It reads a little nicer, but more a matter of preference.